### DISCUSSION TITLE
Set-env and add-path deprecation warning

### DISCUSSION THREAD

**Discussion by reikdas** [community/community#25447] [Actions]
<p>I am not using <code>set-env</code> or <code>add-path</code> in my ci.yml but I am still getting warnings that they are soon going to be deprecated.</p>
<p>For eg here - <a href="https://github.com/reikdas/cling/actions/runs/309226905" rel="noopener nofollow ugc">https://github.com/reikdas/cling/actions/runs/309226905</a></p>
<p>This is my ci.yml - <a href="https://github.com/reikdas/cling/actions/runs/309226905/workflow" rel="noopener nofollow ugc">https://github.com/reikdas/cling/actions/runs/309226905/workflow</a></p>
<p>How do I fix this?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I see two possible causes:</p>
<ol>
<li>An action you’re using is using those commands. That seems unlikely because the only action you’re using is <code>actions/checkout</code>, but updating it to v2 instead of v1 may be a good idea anyway.</li>
<li>Some script or other tool you’re running is issuing <code>set-env</code> or <code>add-path</code> commands.</li>
</ol>
<p>I noticed the failures affect only the macos builds, and indeed the <code>macports-ci</code> script you’re downloading there contains at least an <code>add-path</code> command. It is also downloading and running additional tools, so those probably cause the other warnings.</p>
<p>You could file an issue against the source repository of that script and hope for a patch, or find another way to set up macports.</p>

**Comment by Simran-B** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a></p>
<ol>
<li>I disagree. Many actions rely on <code>@actions/core</code>, which implements the function that lets you set environment variables. The recently released v1.2.6 contains a change to use environment files instead, but other actions have yet to update this dependency, including official ones like checkout.</li>
</ol>

**Comment by airtower-luna** [comment]
<p>I’m not seeing any warnings on my workflows that use <code>actions/checkout@v2</code>, so I’m pretty sure it’s not the case there. Unless there’s some code that only runs on Macos or Windows, I wouldn’t notice in that case. :wink:</p>

**Comment by Simran-B** [comment]
<p>What I mean is that many actions rely on <code>@actions/core</code>, and a lot of official actions haven’t had their dependencies updated for quite a while. But you are right that the checkout action doesn’t call <code>exportVariable()</code> or <code>addPath()</code>. My mistake!</p>

**Comment by HunterZ** [comment]
Is `addPath` deprecated or not? I'm confused that it's still in the documentation.
